web-api/src/error/application.rs

```diff
@@ -46,16 +45,30 @@ where
     }
 }
 
+/// Workaround to use a non-constant `tracing::Level` in the `tracing::event!` macro.
+macro_rules! application_event {
+    ($level:expr, $($fields:tt)*) => {{
+        match $level {
+            ::tracing::Level::TRACE => ::tracing::trace!($($fields)*),
+            ::tracing::Level::DEBUG => ::tracing::debug!($($fields)*),
+            ::tracing::Level::INFO => ::tracing::info!($($fields)*),
+            ::tracing::Level::WARN => ::tracing::warn!($($fields)*),
+            ::tracing::Level::ERROR => ::tracing::error!($($fields)*),
```

Comment:
unfortunately we can't use non-constant levels in the `event!` macros. we also can't turn the `ApplicationError::level()` method into an associated constant because of trait object safety.

i wrapped this in a macro to avoid to much bloat, if someone has a nicer solution then i'm happy to change this.